src/Symfony/Bundle/SecurityBundle/Resources/views/Collector/security.html.twig

```diff
@@ -331,8 +306,44 @@
                         </td>
                         <td>{{ profiler_dump(decision.seek('object')) }}</td>
                     </tr>
-                {% endfor %}
-            </tbody>
-        </table>
+                </tbody>
+            </table>
+            {% if collector.voterDetails[key] is not empty %}
+                {% set voter_details_id = 'voter-details-' ~ loop.index %}
+                <a class="btn btn-link text-small sf-toggle" data-toggle-selector="#{{ voter_details_id }}" data-toggle-alt-content="Hide voter details">Show voter details</a>
+                <div id="{{ voter_details_id }}" class="sf-toggle-content sf-toggle-hidden">
+                    <table class="voters">
+                        <thead>
+                            <tr>
+                                <th>#</th>
+                                <th>Voter class</th>
+                                <th>Vote result</th>
+                            </tr>
+                        </thead>
+                        <tbody>
+                        {% for voter_detail in collector.voterDetails[key] %}
+                            <tr>
+                                <td class="font-normal text-small text-muted nowrap">{{ loop.index }}</td>
+                                <td class="font-normal">{{ profiler_dump(voter_detail['class']) }}</td>
+                                <td class="font-normal text-small">
+                                    {% if voter_detail['vote'] is null %}
+                                        -
+                                    {% elseif voter_detail['vote'] == 1 %}
```

Comment:
Checking against the constants here would be more explicit, i.e.,

```
voter_detail['vote'] == constant('Symfony\\Component\\Security\\Core\\Authorization\\Voter\\VoterInterface::ACCESS_GRANTED')
```

Reply:
@rpkamp  It make sense, modified, thanks !